cpp/examples/tutorial_examples/dataset_example.cc

```diff
@@ -75,6 +76,8 @@ arrow::Result<std::string> CreateExampleParquetDataset(
 }
 
 arrow::Status PrepareEnv() {
+  // Initilize the compute module to register the required kernels for Dataset
+  ARROW_RETURN_NOT_OK(arrow::compute::Initialize());
```

Comment:
Should this be done implicitly by the Dataset layer? Or is it not possible? cc @bkietz

Reply:
[arrow::dataset::internal::Initialize](
https://github.com/apache/arrow/blob/a1707dba7c9ec4265c048586c485c526048cf162/cpp/src/arrow/dataset/plan.h#L29) could certainly call `arrow::compute::Initialize`. However that function only registers dataset ExecNode factories, and isn't used here. Implicit initialization is something I'd recommend avoiding until we can provide better guarantees